Ported RitualDismantler

keraldi opened the pull request:
Signed-off-by: tobias <angryaeon@icloud.com>

Commit:
Ported RitualDismantler

Signed-off-by: tobias <angryaeon@icloud.com>

Commit:
Added Hellfire Forge recipe

Destructive Will Crystal, Destructive Will Crystal, Ritual Diviner (default), Weak Blood Shard

Requires a minimum of 500 will and consumes 100 will to craft

keraldi commented:
#1167

TehNut commented on src/main/java/WayofTime/bloodmagic/item/ItemRitualDismantler.java:
```diff
@@ -0,0 +1,81 @@
+        return breakRitualStoneAtMasterStone(player, world, pos, player.getHeldItem(hand));
+    }
+
+    public EnumActionResult breakRitualStoneAtMasterStone(EntityPlayer player, World world, BlockPos pos, ItemStack stack) {
```
If you're going to bother moving this to it's own method, it should be static, otherwise just put it back in `onItemUse`.

keraldi replied:
Done.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/ItemRitualDismantler.java:
```diff
@@ -0,0 +1,81 @@
+                }
+            }
+
+            NetworkHelper.getSoulNetwork(player).syphon(SoulTicket.item(stack, 100));
```
Why does it cost 250LP to destroy just the controller, but only 100LP to destroy the entire ritual?

keraldi replied:
My thought process was that the MRS is more powerful. "The entire ritual" does not destroy the MRS.

For cases in which your MRS is above a pool of lava (RitualPump) and you really don't want to destroy the MRS with a pickaxe. Or just because.

In reality, most of the LP values are placeholders for more reasonable proposals.

keraldi replied:
Variable cost up to a maximum for a whole ritual?

Commit:
Changed costs (lazy proposal):
 - 50 LP per normal Ritual Stone
 - discount cost of 200 LP for a whole ritual (smallest ritual has 4 ritual stones)
 - 100 LP for an MRS

keraldi commented:
Done.
